components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/SubscriptionDataHolder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.keymgt;
+
+import org.wso2.carbon.apimgt.keymgt.model.SubscriptionDataStore;
+import org.wso2.carbon.apimgt.keymgt.model.impl.SubscriptionDataStoreImpl;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/*
+ * This class holds tenant wise subscription data stores
+ * */
+public class SubscriptionDataHolder {
+
+    private Map<String, SubscriptionDataStore> subscriptionStore =
+            new ConcurrentHashMap<>();
+    private static SubscriptionDataHolder instance = new SubscriptionDataHolder();
+
+    public static SubscriptionDataHolder getInstance() {
+
+        return instance;
+    }
+
+    public void registerTenantSubscriptionStore(String tenantDomain) {
+
+        SubscriptionDataStore tenantStore = subscriptionStore.get(tenantDomain);
+        if (tenantStore == null) {
+            tenantStore = new SubscriptionDataStoreImpl(tenantDomain);
+        }
+        subscriptionStore.put(tenantDomain, tenantStore);
+    }
+
+    public void unregisterTenantSubscriptionStore(int tenantId) {
+
+        subscriptionStore.remove(tenantId);
+    }
+
+    public SubscriptionDataStore getTenantSubscriptionStore(int tenantId) {
+
+        return subscriptionStore.get(tenantId);
```

Comment:
here we are retrieving using the tenant id . but we set the map using the tenant domain. need to use same way